From e24eb7e064d0c68962cee4a09de3a54e40e93130 Mon Sep 17 00:00:00 2001
From: Ivan Gomez Castellanos <ivan.gomez@ti.com>
Date: Thu, 25 Mar 2010 15:38:31 -0600
Subject: [PATCH 093/135] DSPBRIDGE: Remove mem.c and mem.h files

These files are not used anymore.

Also, a cleanup of memdefs.h file is done, because the Memory Pool
Attributes are not used after removing mem.c and mem.h files.

Signed-off-by: Ivan Gomez Castellanos <ivan.gomez@ti.com>
---
 arch/arm/plat-omap/include/dspbridge/mem.h      |  142 -----------------------
 arch/arm/plat-omap/include/dspbridge/memdefs.h  |    7 -
 arch/arm/plat-omap/include/dspbridge/wcdioctl.h |   29 -----
 drivers/dsp/bridge/Makefile                     |    6 +-
 drivers/dsp/bridge/gen/gs.c                     |    4 +-
 drivers/dsp/bridge/pmgr/chnl.c                  |    1 -
 drivers/dsp/bridge/pmgr/cmm.c                   |    1 -
 drivers/dsp/bridge/pmgr/cod.c                   |    1 -
 drivers/dsp/bridge/pmgr/dbll.c                  |    1 -
 drivers/dsp/bridge/pmgr/dev.c                   |    1 -
 drivers/dsp/bridge/pmgr/dmm.c                   |    1 -
 drivers/dsp/bridge/pmgr/io.c                    |    1 -
 drivers/dsp/bridge/pmgr/msg.c                   |    3 -
 drivers/dsp/bridge/pmgr/wcd.c                   |    1 -
 drivers/dsp/bridge/rmgr/dbdcd.c                 |    3 -
 drivers/dsp/bridge/rmgr/disp.c                  |    1 -
 drivers/dsp/bridge/rmgr/drv.c                   |    1 -
 drivers/dsp/bridge/rmgr/dspdrv.c                |    1 -
 drivers/dsp/bridge/rmgr/mgr.c                   |    1 -
 drivers/dsp/bridge/rmgr/nldr.c                  |    3 -
 drivers/dsp/bridge/rmgr/node.c                  |    2 +-
 drivers/dsp/bridge/rmgr/proc.c                  |    1 -
 drivers/dsp/bridge/rmgr/rmm.c                   |    1 -
 drivers/dsp/bridge/rmgr/strm.c                  |    1 -
 drivers/dsp/bridge/services/clk.c               |    3 -
 drivers/dsp/bridge/services/mem.c               |   98 ----------------
 drivers/dsp/bridge/services/services.c          |    1 -
 drivers/dsp/bridge/wmd/chnl_sm.c                |    1 -
 drivers/dsp/bridge/wmd/io_sm.c                  |    1 -
 drivers/dsp/bridge/wmd/mmu_fault.c              |    1 -
 drivers/dsp/bridge/wmd/msg_sm.c                 |    1 -
 drivers/dsp/bridge/wmd/tiomap3430.c             |    1 -
 drivers/dsp/bridge/wmd/tiomap3430_pwr.c         |    3 -
 drivers/dsp/bridge/wmd/tiomap_io.c              |    1 -
 drivers/dsp/bridge/wmd/ue_deh.c                 |    1 -
 35 files changed, 5 insertions(+), 321 deletions(-)
 delete mode 100644 arch/arm/plat-omap/include/dspbridge/mem.h
 delete mode 100644 drivers/dsp/bridge/services/mem.c

diff --git a/arch/arm/plat-omap/include/dspbridge/mem.h b/arch/arm/plat-omap/include/dspbridge/mem.h
deleted file mode 100644
index 854945a..0000000
--- a/arch/arm/plat-omap/include/dspbridge/mem.h
+++ /dev/null
@@ -1,142 +0,0 @@
-/*
- * mem.h
- *
- * DSP-BIOS Bridge driver support functions for TI OMAP processors.
- *
- * Memory management and address mapping services for the DSP/BIOS Bridge
- * class driver and mini-driver.
- *
- * Copyright (C) 2008 Texas Instruments, Inc.
- *
- * This package is free software; you can redistribute it and/or modify
- * it under the terms of the GNU General Public License version 2 as
- * published by the Free Software Foundation.
- *
- * THIS PACKAGE IS PROVIDED ``AS IS'' AND WITHOUT ANY EXPRESS OR
- * IMPLIED WARRANTIES, INCLUDING, WITHOUT LIMITATION, THE IMPLIED
- * WARRANTIES OF MERCHANTIBILITY AND FITNESS FOR A PARTICULAR PURPOSE.
- */
-
-#ifndef MEM_
-#define MEM_
-
-#include <dspbridge/host_os.h>
-#include <dspbridge/memdefs.h>
-
-/*
- *  ======== mem_alloc ========
- *  Purpose:
- *      Allocate memory from the paged or non-paged pools.
- *  Parameters:
- *      byte_size: Number of bytes to allocate.
- *      type:   Type of memory to allocate; one of:
- *              MEM_PAGED: Allocate from pageable memory.
- *              MEM_NONPAGED: Allocate from page locked memory.
- *  Returns:
- *      Pointer to a block of memory;
- *      NULL if memory couldn't be allocated, if byte_size == 0, or if type is
- *      not one of MEM_PAGED or MEM_NONPAGED.
- *  Requires:
- *      MEM initialized.
- *  Ensures:
- *      The returned pointer, if not NULL, points to a valid memory block of
- *      the size requested.
- */
-extern void *mem_alloc(IN u32 byte_size, IN enum mem_poolattrs type);
-
-/*
- *  ======== MEM_ALLOC_OBJECT ========
- *  Purpose:
- *      Allocate an object, and set it's signature.
- *  Parameters:
- *      pObj:       Pointer to the new object.
- *      Obj:        Type of the object to allocate.
- *      Signature:  Magic field value.  Must be non-zero.
- *  Returns:
- *  Requires:
- *      Same requirements as mem_calloc(); and
- *      The object structure has a dw_signature field.  The compiler ensures
- *      this requirement.
- *  Ensures:
- *      A subsequent call to MEM_IS_VALID_HANDLE() will succeed for this object.
- */
-#define MEM_ALLOC_OBJECT(pObj, Obj, Signature)           \
-{                                                       \
-    pObj = kzalloc(sizeof(Obj), GFP_KERNEL);		\
-    if (pObj) {                                         \
-	pObj->dw_signature = Signature;                  \
-    }                                                   \
-}
-
-/*
- *  ======== mem_calloc ========
- *  Purpose:
- *      Allocate zero-initialized memory from the paged or non-paged pools.
- *  Parameters:
- *      byte_size: Number of bytes to allocate.
- *      type:   Type of memory to allocate; one of:
- *              MEM_PAGED:   Allocate from pageable memory.
- *              MEM_NONPAGED: Allocate from page locked memory.
- *  Returns:
- *      Pointer to a block of zeroed memory;
- *      NULL if memory couldn't be allocated, if byte_size == 0, or if type is
- *      not one of MEM_PAGED or MEM_NONPAGED.
- *  Requires:
- *      MEM initialized.
- *  Ensures:
- *      The returned pointer, if not NULL, points to a valid memory block
- *      of the size requested.
- */
-extern void *mem_calloc(IN u32 byte_size, IN enum mem_poolattrs type);
-
-/*
- *  ======== MEM_FREE_OBJECT ========
- *  Purpose:
- *      Utility macro to invalidate an object's signature, and deallocate it.
- *  Parameters:
- *      pObj:   Pointer to the object to free.
- *  Returns:
- *  Requires:
- *      Same requirements as kfree().
- *  Ensures:
- *      A subsequent call to MEM_IS_VALID_HANDLE() will fail for this object.
- */
-#define MEM_FREE_OBJECT(pObj)    \
-{                               \
-    pObj->dw_signature = 0x00;   \
-    kfree(pObj);                \
-}
-
-/*
- *  ======== mem_get_num_pages ========
- *  Purpose:
- *      Calculate the number of pages corresponding to the supplied buffer.
- *  Parameters:
- *      paddr:  Linear (virtual) address of the buffer.
- *      byte_size: Number of bytes in the buffer.
- *  Returns:
- *      Number of pages.
- *  Requires:
- *      MEM initialized.
- *  Ensures:
- *      If byte_size > 0, number of pages returned > 0.
- */
-extern s32 mem_get_num_pages(IN void *paddr, IN u32 byte_size);
-
-/*
- *  ======== MEM_IS_VALID_HANDLE ========
- *  Purpose:
- *      Validate the object handle.
- *  Parameters:
- *      hObj:   Handle to object created with MEM_ALLOC_OBJECT().
- *      Sig:    Expected signature u32.
- *  Returns:
- *      TRUE if handle is valid; FALSE otherwise.
- *  Requires:
- *      The object structure has a dw_signature field. Ensured by compiler.
- *  Ensures:
- */
-#define MEM_IS_VALID_HANDLE(hObj, Sig)                \
-     ((hObj != NULL) && (hObj->dw_signature == Sig))
-
-#endif /* MEM_ */
diff --git a/arch/arm/plat-omap/include/dspbridge/memdefs.h b/arch/arm/plat-omap/include/dspbridge/memdefs.h
index 23885fd..cce6957 100644
--- a/arch/arm/plat-omap/include/dspbridge/memdefs.h
+++ b/arch/arm/plat-omap/include/dspbridge/memdefs.h
@@ -19,13 +19,6 @@
 #ifndef MEMDEFS_
 #define MEMDEFS_
 
-/* Memory Pool Attributes: */
-enum mem_poolattrs {
-	MEM_PAGED = 0,
-	MEM_NONPAGED = 1,
-	MEM_LARGEVIRTMEM = 2
-};
-
 /*
  *  MEM_VIRTUALSEGID is used by Node & Strm to access virtual address space in
  *  the correct client process context.
diff --git a/arch/arm/plat-omap/include/dspbridge/wcdioctl.h b/arch/arm/plat-omap/include/dspbridge/wcdioctl.h
index b6a4dda..49bec37 100644
--- a/arch/arm/plat-omap/include/dspbridge/wcdioctl.h
+++ b/arch/arm/plat-omap/include/dspbridge/wcdioctl.h
@@ -20,7 +20,6 @@
 #ifndef WCDIOCTL_
 #define WCDIOCTL_
 
-#include <dspbridge/mem.h>
 #include <dspbridge/cmm.h>
 #include <dspbridge/strmdefs.h>
 #include <dspbridge/dbdcd.h>
@@ -352,34 +351,6 @@ union Trapped_Args {
 		struct cmm_info __user *cmm_info_obj;
 	} args_cmm_getinfo;
 
-	/* MEM Module */
-	struct {
-		u32 byte_size;
-		enum mem_poolattrs type;
-		void *mem;
-	} args_mem_alloc;
-
-	struct {
-		u32 byte_size;
-		enum mem_poolattrs type;
-		void *mem;
-	} args_mem_calloc;
-
-	struct {
-		void *mem;
-	} args_mem_free;
-
-	struct {
-		void *pbuffer;
-		u32 size;
-		void *locked_buffer;
-	} args_mem_pagelock;
-
-	struct {
-		void *pbuffer;
-		u32 size;
-	} args_mem_pageunlock;
-
 	/* UTIL module */
 	struct {
 		s32 util_argc;
diff --git a/drivers/dsp/bridge/Makefile b/drivers/dsp/bridge/Makefile
index e570a59..e243670 100644
--- a/drivers/dsp/bridge/Makefile
+++ b/drivers/dsp/bridge/Makefile
@@ -1,10 +1,8 @@
 obj-$(CONFIG_MPU_BRIDGE)	+= bridgedriver.o
 
 libgen = gen/gb.o gen/gs.o gen/gh.o gen/uuidutil.o
-libservices = services/mem.o services/sync.o \
-		services/clk.o services/cfg.o \
-               services/ntfy.o \
-		services/services.o
+libservices = services/sync.o services/clk.o services/cfg.o \
+		services/ntfy.o services/services.o
 libwmd = wmd/chnl_sm.o wmd/msg_sm.o wmd/io_sm.o wmd/tiomap3430.o \
 		wmd/tiomap3430_pwr.o wmd/tiomap_io.o \
 		wmd/mmu_fault.o wmd/ue_deh.o wmd/wdt.o
diff --git a/drivers/dsp/bridge/gen/gs.c b/drivers/dsp/bridge/gen/gs.c
index 1a43fd5..3d091b9 100644
--- a/drivers/dsp/bridge/gen/gs.c
+++ b/drivers/dsp/bridge/gen/gs.c
@@ -20,12 +20,12 @@
 #include <dspbridge/std.h>
 #include <dspbridge/dbdefs.h>
 #include <linux/types.h>
-/*  ----------------------------------- OS Adaptation Layer */
-#include <dspbridge/mem.h>
 
 /*  ----------------------------------- This */
 #include <dspbridge/gs.h>
 
+#include <linux/slab.h>
+
 /*  ----------------------------------- Globals */
 static u32 cumsize;
 
diff --git a/drivers/dsp/bridge/pmgr/chnl.c b/drivers/dsp/bridge/pmgr/chnl.c
index 41834fb..ea41ccf 100644
--- a/drivers/dsp/bridge/pmgr/chnl.c
+++ b/drivers/dsp/bridge/pmgr/chnl.c
@@ -30,7 +30,6 @@
 
 /*  ----------------------------------- OS Adaptation Layer */
 #include <dspbridge/cfg.h>
-#include <dspbridge/mem.h>
 #include <dspbridge/sync.h>
 
 /*  ----------------------------------- Platform Manager */
diff --git a/drivers/dsp/bridge/pmgr/cmm.c b/drivers/dsp/bridge/pmgr/cmm.c
index cf5c9e1..cd10c97 100644
--- a/drivers/dsp/bridge/pmgr/cmm.c
+++ b/drivers/dsp/bridge/pmgr/cmm.c
@@ -41,7 +41,6 @@
 /*  ----------------------------------- OS Adaptation Layer */
 #include <dspbridge/cfg.h>
 #include <dspbridge/list.h>
-#include <dspbridge/mem.h>
 #include <dspbridge/sync.h>
 #include <dspbridge/utildefs.h>
 
diff --git a/drivers/dsp/bridge/pmgr/cod.c b/drivers/dsp/bridge/pmgr/cod.c
index 014e23f..5d9f085 100644
--- a/drivers/dsp/bridge/pmgr/cod.c
+++ b/drivers/dsp/bridge/pmgr/cod.c
@@ -35,7 +35,6 @@
 
 /*  ----------------------------------- OS Adaptation Layer */
 #include <dspbridge/ldr.h>
-#include <dspbridge/mem.h>
 
 /*  ----------------------------------- Platform Manager */
 /* Include appropriate loader header file */
diff --git a/drivers/dsp/bridge/pmgr/dbll.c b/drivers/dsp/bridge/pmgr/dbll.c
index baa0366..c535252 100644
--- a/drivers/dsp/bridge/pmgr/dbll.c
+++ b/drivers/dsp/bridge/pmgr/dbll.c
@@ -27,7 +27,6 @@
 #include <dspbridge/gh.h>
 
 /*  ----------------------------------- OS Adaptation Layer */
-#include <dspbridge/mem.h>
 
 /* Dynamic loader library interface */
 #include <dspbridge/dynamic_loader.h>
diff --git a/drivers/dsp/bridge/pmgr/dev.c b/drivers/dsp/bridge/pmgr/dev.c
index a0ee17a..2bb74f1 100644
--- a/drivers/dsp/bridge/pmgr/dev.c
+++ b/drivers/dsp/bridge/pmgr/dev.c
@@ -31,7 +31,6 @@
 #include <dspbridge/cfg.h>
 #include <dspbridge/ldr.h>
 #include <dspbridge/list.h>
-#include <dspbridge/mem.h>
 
 /*  ----------------------------------- Platform Manager */
 #include <dspbridge/cod.h>
diff --git a/drivers/dsp/bridge/pmgr/dmm.c b/drivers/dsp/bridge/pmgr/dmm.c
index f0b4688..d0f0c98 100644
--- a/drivers/dsp/bridge/pmgr/dmm.c
+++ b/drivers/dsp/bridge/pmgr/dmm.c
@@ -33,7 +33,6 @@
 #include <dspbridge/errbase.h>
 
 /*  ----------------------------------- OS Adaptation Layer */
-#include <dspbridge/mem.h>
 #include <dspbridge/sync.h>
 
 /*  ----------------------------------- Platform Manager */
diff --git a/drivers/dsp/bridge/pmgr/io.c b/drivers/dsp/bridge/pmgr/io.c
index 154ed67..e16d47b 100644
--- a/drivers/dsp/bridge/pmgr/io.c
+++ b/drivers/dsp/bridge/pmgr/io.c
@@ -29,7 +29,6 @@
 
 /*  ----------------------------------- OS Adaptation Layer */
 #include <dspbridge/cfg.h>
-#include <dspbridge/mem.h>
 
 /*  ----------------------------------- Platform Manager */
 #include <dspbridge/dev.h>
diff --git a/drivers/dsp/bridge/pmgr/msg.c b/drivers/dsp/bridge/pmgr/msg.c
index 04dc540..78bfb46 100644
--- a/drivers/dsp/bridge/pmgr/msg.c
+++ b/drivers/dsp/bridge/pmgr/msg.c
@@ -27,9 +27,6 @@
 /*  ----------------------------------- Trace & Debug */
 #include <dspbridge/dbc.h>
 
-/*  ----------------------------------- OS Adaptation Layer */
-#include <dspbridge/mem.h>
-
 /*  ----------------------------------- Mini Driver */
 #include <dspbridge/wmd.h>
 
diff --git a/drivers/dsp/bridge/pmgr/wcd.c b/drivers/dsp/bridge/pmgr/wcd.c
index 0c54acf..7c32fef 100644
--- a/drivers/dsp/bridge/pmgr/wcd.c
+++ b/drivers/dsp/bridge/pmgr/wcd.c
@@ -30,7 +30,6 @@
 
 /*  ----------------------------------- OS Adaptation Layer */
 #include <dspbridge/cfg.h>
-#include <dspbridge/mem.h>
 #include <dspbridge/ntfy.h>
 #include <dspbridge/services.h>
 
diff --git a/drivers/dsp/bridge/rmgr/dbdcd.c b/drivers/dsp/bridge/rmgr/dbdcd.c
index 525a03a..af53286 100644
--- a/drivers/dsp/bridge/rmgr/dbdcd.c
+++ b/drivers/dsp/bridge/rmgr/dbdcd.c
@@ -33,9 +33,6 @@
 /*  ----------------------------------- Trace & Debug */
 #include <dspbridge/dbc.h>
 
-/*  ----------------------------------- OS Adaptation Layer */
-#include <dspbridge/mem.h>
-
 /*  ----------------------------------- Platform Manager */
 #include <dspbridge/cod.h>
 
diff --git a/drivers/dsp/bridge/rmgr/disp.c b/drivers/dsp/bridge/rmgr/disp.c
index 3a8ea5d..665ff56 100644
--- a/drivers/dsp/bridge/rmgr/disp.c
+++ b/drivers/dsp/bridge/rmgr/disp.c
@@ -29,7 +29,6 @@
 #include <dspbridge/dbc.h>
 
 /*  ----------------------------------- OS Adaptation Layer */
-#include <dspbridge/mem.h>
 #include <dspbridge/sync.h>
 
 /*  ----------------------------------- Link Driver */
diff --git a/drivers/dsp/bridge/rmgr/drv.c b/drivers/dsp/bridge/rmgr/drv.c
index d155c57..d79aa57 100644
--- a/drivers/dsp/bridge/rmgr/drv.c
+++ b/drivers/dsp/bridge/rmgr/drv.c
@@ -30,7 +30,6 @@
 /*  ----------------------------------- OS Adaptation Layer */
 #include <dspbridge/cfg.h>
 #include <dspbridge/list.h>
-#include <dspbridge/mem.h>
 
 /*  ----------------------------------- This */
 #include <dspbridge/drv.h>
diff --git a/drivers/dsp/bridge/rmgr/dspdrv.c b/drivers/dsp/bridge/rmgr/dspdrv.c
index 109ae1d..efe6e26 100644
--- a/drivers/dsp/bridge/rmgr/dspdrv.c
+++ b/drivers/dsp/bridge/rmgr/dspdrv.c
@@ -29,7 +29,6 @@
 
 /*  ----------------------------------- OS Adaptation Layer */
 #include <dspbridge/cfg.h>
-#include <dspbridge/mem.h>
 
 /*  ----------------------------------- Platform Manager */
 #include <dspbridge/drv.h>
diff --git a/drivers/dsp/bridge/rmgr/mgr.c b/drivers/dsp/bridge/rmgr/mgr.c
index cbabfb6..78d80af 100644
--- a/drivers/dsp/bridge/rmgr/mgr.c
+++ b/drivers/dsp/bridge/rmgr/mgr.c
@@ -28,7 +28,6 @@
 
 /*  ----------------------------------- OS Adaptation Layer */
 #include <dspbridge/cfg.h>
-#include <dspbridge/mem.h>
 #include <dspbridge/sync.h>
 
 /*  ----------------------------------- Others */
diff --git a/drivers/dsp/bridge/rmgr/nldr.c b/drivers/dsp/bridge/rmgr/nldr.c
index e984faf..90eadc0 100644
--- a/drivers/dsp/bridge/rmgr/nldr.c
+++ b/drivers/dsp/bridge/rmgr/nldr.c
@@ -24,9 +24,6 @@
 
 #include <dspbridge/dbc.h>
 
-/* OS adaptation layer */
-#include <dspbridge/mem.h>
-
 /* Platform manager */
 #include <dspbridge/cod.h>
 #include <dspbridge/dev.h>
diff --git a/drivers/dsp/bridge/rmgr/node.c b/drivers/dsp/bridge/rmgr/node.c
index 9b26d23..db2d67e 100644
--- a/drivers/dsp/bridge/rmgr/node.c
+++ b/drivers/dsp/bridge/rmgr/node.c
@@ -30,7 +30,7 @@
 /*  ----------------------------------- OS Adaptation Layer */
 #include <dspbridge/cfg.h>
 #include <dspbridge/list.h>
-#include <dspbridge/mem.h>
+#include <dspbridge/memdefs.h>
 #include <dspbridge/proc.h>
 #include <dspbridge/strm.h>
 #include <dspbridge/sync.h>
diff --git a/drivers/dsp/bridge/rmgr/proc.c b/drivers/dsp/bridge/rmgr/proc.c
index 4e8cf80..9b025b2 100644
--- a/drivers/dsp/bridge/rmgr/proc.c
+++ b/drivers/dsp/bridge/rmgr/proc.c
@@ -30,7 +30,6 @@
 /*  ----------------------------------- OS Adaptation Layer */
 #include <dspbridge/cfg.h>
 #include <dspbridge/list.h>
-#include <dspbridge/mem.h>
 #include <dspbridge/ntfy.h>
 #include <dspbridge/sync.h>
 /*  ----------------------------------- Mini Driver */
diff --git a/drivers/dsp/bridge/rmgr/rmm.c b/drivers/dsp/bridge/rmgr/rmm.c
index c50d98c..ab1aa9d 100644
--- a/drivers/dsp/bridge/rmgr/rmm.c
+++ b/drivers/dsp/bridge/rmgr/rmm.c
@@ -47,7 +47,6 @@
 
 /*  ----------------------------------- OS Adaptation Layer */
 #include <dspbridge/list.h>
-#include <dspbridge/mem.h>
 
 /*  ----------------------------------- This */
 #include <dspbridge/rmm.h>
diff --git a/drivers/dsp/bridge/rmgr/strm.c b/drivers/dsp/bridge/rmgr/strm.c
index 467d0e4..a9a1e5b 100644
--- a/drivers/dsp/bridge/rmgr/strm.c
+++ b/drivers/dsp/bridge/rmgr/strm.c
@@ -28,7 +28,6 @@
 #include <dspbridge/dbc.h>
 
 /*  ----------------------------------- OS Adaptation Layer */
-#include <dspbridge/mem.h>
 #include <dspbridge/sync.h>
 
 /*  ----------------------------------- Mini Driver */
diff --git a/drivers/dsp/bridge/services/clk.c b/drivers/dsp/bridge/services/clk.c
index cb383ef..fb2ca81 100644
--- a/drivers/dsp/bridge/services/clk.c
+++ b/drivers/dsp/bridge/services/clk.c
@@ -27,9 +27,6 @@
 /*  ----------------------------------- Trace & Debug */
 #include <dspbridge/dbc.h>
 
-/*  ----------------------------------- OS Adaptation Layer */
-#include <dspbridge/mem.h>
-
 /*  ----------------------------------- This */
 #include <dspbridge/clk.h>
 
diff --git a/drivers/dsp/bridge/services/mem.c b/drivers/dsp/bridge/services/mem.c
deleted file mode 100644
index 1d3f32d..0000000
--- a/drivers/dsp/bridge/services/mem.c
+++ /dev/null
@@ -1,98 +0,0 @@
-/*
- * mem.c
- *
- * DSP-BIOS Bridge driver support functions for TI OMAP processors.
- *
- * Implementation of platform specific memory services.
- *
- * Copyright (C) 2005-2006 Texas Instruments, Inc.
- *
- * This package is free software; you can redistribute it and/or modify
- * it under the terms of the GNU General Public License version 2 as
- * published by the Free Software Foundation.
- *
- * THIS PACKAGE IS PROVIDED ``AS IS'' AND WITHOUT ANY EXPRESS OR
- * IMPLIED WARRANTIES, INCLUDING, WITHOUT LIMITATION, THE IMPLIED
- * WARRANTIES OF MERCHANTIBILITY AND FITNESS FOR A PARTICULAR PURPOSE.
- */
-
-/*  ----------------------------------- Host OS */
-#include <dspbridge/host_os.h>
-
-/*  ----------------------------------- DSP/BIOS Bridge */
-#include <dspbridge/std.h>
-#include <dspbridge/dbdefs.h>
-#include <dspbridge/errbase.h>
-
-/*  ----------------------------------- Trace & Debug */
-#include <dspbridge/dbc.h>
-
-/*  ----------------------------------- This */
-#include <dspbridge/mem.h>
-#include <dspbridge/list.h>
-
-/*  ----------------------------------- Defines */
-#define MEM512MB   0x1fffffff
-
-/*  ----------------------------------- Globals */
-
-
-/*
- *  ======== mem_alloc ========
- *  Purpose:
- *      Allocate memory from the paged or non-paged pools.
- */
-void *mem_alloc(u32 byte_size, enum mem_poolattrs type)
-{
-	void *mem = NULL;
-
-	if (byte_size > 0) {
-		switch (type) {
-		case MEM_NONPAGED:
-			/* Fall through */
-		case MEM_PAGED:
-			mem = kmalloc(byte_size,
-				      (in_atomic()) ? GFP_ATOMIC : GFP_KERNEL);
-			break;
-		case MEM_LARGEVIRTMEM:
-			mem = vmalloc(byte_size);
-			break;
-
-		default:
-			break;
-		}
-	}
-
-	return mem;
-}
-
-/*
- *  ======== mem_calloc ========
- *  Purpose:
- *      Allocate zero-initialized memory from the paged or non-paged pools.
- */
-void *mem_calloc(u32 byte_size, enum mem_poolattrs type)
-{
-	void *mem = NULL;
-
-	if (byte_size > 0) {
-		switch (type) {
-		case MEM_NONPAGED:
-			/*  Fall through */
-		case MEM_PAGED:
-			mem = kzalloc(byte_size,
-				      (in_atomic()) ? GFP_ATOMIC : GFP_KERNEL);
-			break;
-		case MEM_LARGEVIRTMEM:
-			mem = __vmalloc(byte_size,
-					GFP_KERNEL | __GFP_HIGHMEM | __GFP_ZERO,
-					PAGE_KERNEL);
-			break;
-		default:
-			break;
-		}
-	}
-
-	return mem;
-}
-
diff --git a/drivers/dsp/bridge/services/services.c b/drivers/dsp/bridge/services/services.c
index d6de172..19ab4d0 100644
--- a/drivers/dsp/bridge/services/services.c
+++ b/drivers/dsp/bridge/services/services.c
@@ -27,7 +27,6 @@
 
 /*  ----------------------------------- OS Adaptation Layer */
 #include <dspbridge/cfg.h>
-#include <dspbridge/mem.h>
 #include <dspbridge/ntfy.h>
 #include <dspbridge/sync.h>
 #include <dspbridge/clk.h>
diff --git a/drivers/dsp/bridge/wmd/chnl_sm.c b/drivers/dsp/bridge/wmd/chnl_sm.c
index 758899b..03e9fbe 100644
--- a/drivers/dsp/bridge/wmd/chnl_sm.c
+++ b/drivers/dsp/bridge/wmd/chnl_sm.c
@@ -54,7 +54,6 @@
 #include <dspbridge/dbc.h>
 
 /*  ----------------------------------- OS Adaptation Layer */
-#include <dspbridge/mem.h>
 #include <dspbridge/cfg.h>
 #include <dspbridge/sync.h>
 
diff --git a/drivers/dsp/bridge/wmd/io_sm.c b/drivers/dsp/bridge/wmd/io_sm.c
index a787724..127c7bc 100644
--- a/drivers/dsp/bridge/wmd/io_sm.c
+++ b/drivers/dsp/bridge/wmd/io_sm.c
@@ -38,7 +38,6 @@
 
 /* Services Layer */
 #include <dspbridge/cfg.h>
-#include <dspbridge/mem.h>
 #include <dspbridge/ntfy.h>
 #include <dspbridge/sync.h>
 
diff --git a/drivers/dsp/bridge/wmd/mmu_fault.c b/drivers/dsp/bridge/wmd/mmu_fault.c
index 5d6ce8b..375d7ec 100644
--- a/drivers/dsp/bridge/wmd/mmu_fault.c
+++ b/drivers/dsp/bridge/wmd/mmu_fault.c
@@ -26,7 +26,6 @@
 #include <dspbridge/dbc.h>
 
 /*  ----------------------------------- OS Adaptation Layer */
-#include <dspbridge/mem.h>
 #include <dspbridge/drv.h>
 
 /*  ----------------------------------- Link Driver */
diff --git a/drivers/dsp/bridge/wmd/msg_sm.c b/drivers/dsp/bridge/wmd/msg_sm.c
index eb835a1..83e9be5 100644
--- a/drivers/dsp/bridge/wmd/msg_sm.c
+++ b/drivers/dsp/bridge/wmd/msg_sm.c
@@ -26,7 +26,6 @@
 
 /*  ----------------------------------- OS Adaptation Layer */
 #include <dspbridge/list.h>
-#include <dspbridge/mem.h>
 #include <dspbridge/sync.h>
 
 /*  ----------------------------------- Platform Manager */
diff --git a/drivers/dsp/bridge/wmd/tiomap3430.c b/drivers/dsp/bridge/wmd/tiomap3430.c
index 4a02ff5..a69f663 100644
--- a/drivers/dsp/bridge/wmd/tiomap3430.c
+++ b/drivers/dsp/bridge/wmd/tiomap3430.c
@@ -31,7 +31,6 @@
 #include <dspbridge/dbc.h>
 
 /*  ----------------------------------- OS Adaptation Layer */
-#include <dspbridge/mem.h>
 #include <dspbridge/cfg.h>
 #include <dspbridge/drv.h>
 #include <dspbridge/sync.h>
diff --git a/drivers/dsp/bridge/wmd/tiomap3430_pwr.c b/drivers/dsp/bridge/wmd/tiomap3430_pwr.c
index d33d713..87ec8d2 100644
--- a/drivers/dsp/bridge/wmd/tiomap3430_pwr.c
+++ b/drivers/dsp/bridge/wmd/tiomap3430_pwr.c
@@ -23,9 +23,6 @@
 #include <dspbridge/drv.h>
 #include <dspbridge/io_sm.h>
 
-/*  ----------------------------------- OS Adaptation Layer */
-#include <dspbridge/mem.h>
-
 /*  ----------------------------------- Platform Manager */
 #include <dspbridge/brddefs.h>
 #include <dspbridge/dev.h>
diff --git a/drivers/dsp/bridge/wmd/tiomap_io.c b/drivers/dsp/bridge/wmd/tiomap_io.c
index 42240bd..5880428 100644
--- a/drivers/dsp/bridge/wmd/tiomap_io.c
+++ b/drivers/dsp/bridge/wmd/tiomap_io.c
@@ -28,7 +28,6 @@
 #include <dspbridge/drv.h>
 
 /*  ----------------------------------- OS Adaptation Layer */
-#include <dspbridge/mem.h>
 #include <dspbridge/cfg.h>
 #include <dspbridge/wdt.h>
 
diff --git a/drivers/dsp/bridge/wmd/ue_deh.c b/drivers/dsp/bridge/wmd/ue_deh.c
index eb3466f..03b29b6 100644
--- a/drivers/dsp/bridge/wmd/ue_deh.c
+++ b/drivers/dsp/bridge/wmd/ue_deh.c
@@ -30,7 +30,6 @@
 
 /*  ----------------------------------- OS Adaptation Layer */
 #include <dspbridge/cfg.h>
-#include <dspbridge/mem.h>
 #include <dspbridge/ntfy.h>
 #include <dspbridge/drv.h>
 
-- 
1.7.0.4

